src/EnergyPlus/GlobalNames.cc

```diff
@@ -334,6 +358,41 @@ namespace GlobalNames {
 
 	}
 
+	void
+	VerifyUniqueADUName(
+		std::string const & TypeToVerify,
+		std::string const & NameToVerify,
+		bool & ErrorFound,
+		std::string const & StringToDisplay
+	)
+	{
+
+
+		ErrorFound = false;
+		int Found = 0;
+
+		if ( numAirDistUnits > 0 ) Found = FindItemInList( NameToVerify, aDUNames, &ComponentNameData::CompName, numAirDistUnits );
+
+		if ( Found != 0 ) {
+			ShowSevereError( StringToDisplay + ", duplicate name=" + NameToVerify + ", ADU Type=\"" + aDUNames( Found ).CompType + "\"" );
+			ShowContinueError( "...Current entry is Air Distribution Unit Type=\"" + TypeToVerify + "\"." );
+			ErrorFound = true;
+		} else {
+			if ( numAirDistUnits == 0 ) {
+				curMaxAirDistUnits = 4;
+				aDUNames.allocate( curMaxAirDistUnits );
+			} else if ( numAirDistUnits == curMaxAirDistUnits ) {
+				curMaxAirDistUnits += 4;
+				aDUNames.redimension( curMaxAirDistUnits );
```

Comment:
@EnergyArchmage For new code, i'd suggest moving to a `push_back()` approach instead of `redimension()` since @DeadParrot added that functionality earlier this summer.

```cpp
ComponentNameData aDUdata( MakeUPPERCase( TypeToVerify ), NameToVerify );

// This has a member constructor that works but if not then you do this...
// aDUdata.CompType = MakeUPPERCase( TypeToVerify );
// aDUdata.CompName = NameToVerify;

aDUNames.push_back( aDUdata );
// could use aDUNames.emplace_back( MakeUPPERCase( TypeToVerify ), NameToVerify );
// this would avoid both lines above and is better.

++numAirDistUnits;
```
This code snippet would replace everything in the `else` statement. This is a MUCH better approach than the old Fortran type approach. It is better for performance (speed and memory) and readability.

@DeadParrot As an aside, i've had some trouble with emplace_back() when i've played around with it locally. Sometimes it doesn't work, but I can see if I can find a concrete example to send to you.